Sources/ApodiniDatabase/PropertyVisitors/VisitableFieldProperty.swift

```diff
@@ -0,0 +1,57 @@
+import Fluent
+
+/// A protocol to make the `FieldProperty` of a database model accessible at run time.
+/// It returns an `AnyCodable` object which contains the value associated with this `FieldProperty`
+protocol VisitableFieldProperty {
+    func accept<Visitor: FieldPropertyVisitor>(_ visitor: Visitor) -> Visitor.Value where Visitor.Value == TypeContainer
+}
+
+/// A protocol to make the `FieldProperty` of a database model accessible at run time.
+/// It returns an `AnyCodable` object which contains the value associated with this `IDProperty`
+protocol VisitableIDProperty {
+    func accept<Visitor: IDPropertyVisitor>(_ visitor: Visitor) -> Visitor.Value where Visitor.Value == TypeContainer
+}
+
+/// A protocol all field property visitors have to conform to. It returns whatever has been specified as `Value`.
+protocol FieldPropertyVisitor where Value: Codable {
+    associatedtype Value
+
+    func visit<Model, V>(_ property: FieldProperty<Model, V>) -> Value
+}
+
+/// A protocol all id property visitors have to conform to. It returns whatever has been specified as `Value`.
+protocol IDPropertyVisitor where Value: Codable {
+    associatedtype Value
+    
+    func visit<Model, V>(_ property: IDProperty<Model, V>) -> Value where Model: DatabaseModel
+}
+
+/// A concrete implementation of the `FieldPropertyVisitor` that visits a given `FieldProperty` and returns an `AnyCodable` object
+struct ConcreteTypeVisitor: FieldPropertyVisitor {
+    typealias Value = TypeContainer
+    
+    func visit<Model, V>(_ property: FieldProperty<Model, V>) -> Value where Model: Fields, V: Decodable, V: Encodable {
```

Comment:
You can combine the `V: Decodable, V: Encodable` requirement into a single one. In addition you don't need the `typealias` here if you remove the `associatedType` above.
```suggestion
    func visit<Model, V>(_ property: FieldProperty<Model, V>) -> TypeContainer where Model: Fields, V: Codable {
```